components/Banner/examples/supplemental/MDCBannerView.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+@class MDCButton;
+
+__attribute__((objc_subclassing_restricted)) @interface MDCBannerView : UIView
+
+@property(nonatomic, copy, nonnull) NSString *text;
+
+@property(nonatomic, strong, nullable) UIImage *image;  // default is nil
+
+@property(nonatomic, assign) NSUInteger numberOfButtons;  // default is 1
```

Comment:
I believe assigning this property will change the number of button objects in `buttons`. Would you please document its side-effects? What happens if I change buttons from `2` to `1`? Are all new buttons created with no selectors or are some retained?